contrib/llvm-project/compiler-rt/lib/builtins/clear_cache.c

```diff
@@ -91,8 +91,13 @@ void __clear_cache(void *start, void *end) {
 #elif defined(__mips__) && defined(__OpenBSD__)
   cacheflush(start, (uintptr_t)end - (uintptr_t)start, BCACHE);
 #elif defined(__aarch64__) && !defined(__APPLE__)
+#if defined(__CHERI_PURE_CAPABILITY__)
```

Comment:
We should also apply this to our llvm-project repo. If there are any other builtins that need changes, we might want to import our compiler-rt code as well. If not, the import can probably wait until we port UBSan/MSan/etc to CHERI.

Reply:
There are a few other bits and pieces in Morello's compiler-rt, but nothing that looks urgent to import. I've created https://github.com/CTSRD-CHERI/llvm-project/pull/503 with this bit.

Reply:
I've dropped this change for now and will let @jrtc27 vendor it in.